### PR TITLE
+htp #20002 add authorizeAsync

### PR DESCRIPTION
**Pull request by ktoso**
Resolves this awesome issue nr: #20002, been a while since we had a palindrome nr :-)

And adds an async version for authorizeAsync

### CODE REVIEW

File: akka-http-core/src/main/java/akka/http/javadsl/model/headers/EntityTagRanges.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
```
Comment (johanandren) [inline akka-http-core/src/main/java/akka/http/javadsl/model/headers/EntityTagRanges.java]: Accidental stuff in this commit, or is this related?

Comment (ktoso) [inline akka-http-core/src/main/java/akka/http/javadsl/model/headers/EntityTagRanges.java]: Argh, the other PR got mixed up in this one. I must have messed up stashing, will fix.

**Comment by johanandren** [comment]
Build failure is related to the stuff I think got added to the commit accidentally.

**Comment by johanandren** [comment]
LGTM except for that

**Comment by ktoso** [comment]
Removed the accidental stuff.

**Comment by ktoso** [comment]
Wat, mima touching `akka-http-experimental`? Why u panic there mima, will fix.

**Comment by patriknw** [comment]
LGTM

**Comment by ktoso** [comment]
I disabled MiMa in the `akka-http-experimental` project (as it's experimental still) and will merge after validation.
